### PR TITLE
INGK-851 Implement functionalities to merge dictionaries instances

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Added support for merging dictionary instances. It can only be used for merging COM-KIT and CORE dictionaries.

Fixes # INGK-851

### Type of change

- Implement the add operator for merging dictionary instances.
- Remove the moco_image attribute.

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: ingenialink/dictionary.py [hunk 1]
```diff
@@ -314,6 +332,56 @@ def get_safety_tpdo(self, uid: str) -> DictionarySafetyPDO:
...
+
+        """
+        if not self._is_coco_dictionary(other_dict):
+            self.product_code = other_dict.product_code
+            self.revision_number = other_dict.revision_number
+            self.firmware_version = other_dict.firmware_version
+            self.part_number = other_dict.part_number
+
+    @staticmethod
+    def _is_coco_dictionary(dictionary: "Dictionary") -> bool:
```
Comment (pzinemanas) [inline ingenialink/dictionary.py:381]: Could this be a property instead of a static method?

File: ingenialink/dictionary.py [hunk 2]
```diff
@@ -239,6 +237,26 @@ def __init__(self, dictionary_path: str, interface: Interface) -> None:
...
+        """Merge two dictionary instances.
+
+        It can only be used for merging COM-KIT and CORE dictionaries.
+
+        """
+        if not isinstance(other_dict, type(self)):
+            raise ValueError(
+                f"Cannot merge dictionaries. Expected type: {type(self)}, got: {type(other_dict)}"
+            )
+        if not self._is_coco_dictionary(other_dict) and not self._is_coco_dictionary(self):
```
Comment (pzinemanas) [inline ingenialink/dictionary.py:250]: I'd rather use properties: https://github.com/ingeniamc/ingenialink-python/pull/389/files#r1551055242

**Review by pzinemanas — CHANGES_REQUESTED** [review]
Please address the comments.

File: ingenialink/dictionary.py (hunk 1 above)
Comment (MartinALL91) [reply ingenialink/dictionary.py]: Done

File: ingenialink/dictionary.py (hunk 2 above)
Comment (MartinALL91) [reply ingenialink/dictionary.py]: Done

**Review by pzinemanas — APPROVED** [review]

File: ingenialink/dictionary.py [hunk 3]
```diff
@@ -239,6 +237,26 @@ def __init__(self, dictionary_path: str, interface: Interface) -> None:
...
             raise ILDictionaryParseError("The dictionary is not well-formed.") from e
 
+    def __add__(self, other_dict: "Dictionary") -> "Dictionary":
+        """Merge two dictionary instances.
+
+        It can only be used for merging COM-KIT and CORE dictionaries.
+
+        """
+        if not isinstance(other_dict, type(self)):
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (MartinALL91) [reply ingenialink/dictionary.py]: Done

**Review by WoodElv — APPROVED** [review]

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**